cli-tests/cli-tests.bats

```diff
@@ -554,6 +557,12 @@ _concurrent_modify_worker () {
 }
 
 @test "hard link to symlink" {
+  # On macOS hard links to symlinks actually point to the original file
+  if test $(uname) == "Darwin"
+  then
+    skip "Not applicable on macOS"
```

Comment:
I wonder if this is a case of the macos ln command trying to be clever.

Reply:
Maybe, if you have a FreeBSD machine you could try the same there. The manpage for `ln` on macOS makes explicit FreeBSD references so it may have the same behavior.